### PR TITLE
Added functionality for tracking learning rate

### PR DESCRIPTION
**Pull request by soumik12345**
Fixes WB-NNNN
Fixes #NNNN

Description
-----------
What does the PR do?

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

**Comment by soumik12345** [comment]
cc: @ayulockin

**Comment by ayulockin** [comment]
Thanks for adding this feature. 👯

File: wandb/integration/keras/callbacks/metrics_logger.py [hunk 1]
```diff
@@ -38,10 +38,17 @@ def __init__(self, log_batch_frequency: Optional[int] = None, *args, **kwargs):
             wandb.define_metric("batch/batch_step")
             # set all other batch/ metrics to use this step
             wandb.define_metric("batch/*", step_metric="batch/batch_step")
+    
+    def _get_lr(self, step):
+        try:
+            return self.model.optimizer.learning_rate.numpy().item()
+        except AttributeError:
+            return self.model.optimizer.learning_rate(step=step).numpy()
```
Comment (ayulockin) [inline wandb/integration/keras/callbacks/metrics_logger.py:46]: Passing the step from `on_epoch_end` as `epoch` and from `on_batch_end` as `batch` will give the wrong result IMO (I am confident about this is a LR Scheduler is used).

The learning rate depends on the global iteration, which can be accessed via `self.optimizer.iterations`.

I think the change would be:

`self.model.optimizer.learning_rate(step=self.optimizer.iterations).numpy()`

File: wandb/integration/keras/callbacks/metrics_logger.py [hunk 2]
```diff
@@ -38,10 +38,17 @@ def __init__(self, log_batch_frequency: Optional[int] = None, *args, **kwargs):
...
 
     def on_epoch_end(self, epoch: int, logs: dict = {}):
         """Called at the end of an epoch."""
         wandb.log({"epoch": epoch}, commit=False)
+        logs["learning_rate"] = self._get_lr(step=epoch)
```
Comment (ayulockin) [inline wandb/integration/keras/callbacks/metrics_logger.py:51]: Change accordingly.

File: wandb/integration/keras/callbacks/metrics_logger.py [hunk 3]
```diff
@@ -50,6 +57,7 @@ def on_batch_end(self, batch: int, logs: dict = {}):
             wandb.log({"batch/batch_step": self.global_batch}, commit=False)
 
             logs = {f"batch/{k}": v for k, v in logs.items()}
+            logs["batch/learning_rate"] = self._get_lr(step=batch)
```
Comment (ayulockin) [inline wandb/integration/keras/callbacks/metrics_logger.py:60]: Change accordingly.

**Comment by ayulockin** [comment]
Can you also update the yea test to incorporate learning rate?
